variables.tf

```diff
@@ -0,0 +1,102 @@
+variable "ecr_repository_name" {
+  description = "Name of the ecr repository to hold lambda image"
+  type        = string
+  default     = "my-tf-function"
+}
+
+variable "security_group_id" {
+  description = "id of security group associated with ec2 deployment"
+  type        = string
+  default     = null
+}
+
+variable "public_subnet_ids" {
+  description = "ids of public subnet associated with ec2 deployment"
+  type        = list(string)
+  default     = []
+}
+
+variable "ecr_image_tag_mutability" {
+  description = "Mutability of the ecr image tag"
+  type        = string
+  default     = "MUTABLE"
+}
+
+variable "ecr_scan_on_push" {
+  description = "Describes the actions taken on push"
+  type        = bool
+  default     = true
+}
+
+// not even sure if s3 can have a default name
+variable "s3_tf_bucket_name" {
+  description = "name of the s3 bucket for the lambda with terraform binary"
+  type        = string
+  default     = "terraform-lambda-deploy-state"
```

Comment:
Let's not default this since it's almost guaranteed the bucket name will be taken.